AttributeError: `np.NaN` was removed in the NumPy 2.0 release. Use `np.nan` instead.

Hi, I didn't find an issue open for this on GitHub. This issue started appearing in the GitHub Actions build environment for [xCDAT](https://github.com/xCDAT/xcdat/actions/runs/9650562900/job/26616954254?pr=667) where some of the unit tests call xESMF. The build environment installs `numpy=2.0`, which seems to raise issues with xESMF. Right now we need to constrain `numpy < 2.0` as a workaround to this issue.

### Related Code

https://github.com/pangeo-data/xESMF/blob/530b804c28a9b4f64dd360384897c4c2b34ab8c3/xesmf/smm.py#L229

### Log Output

```python
 xcdat/regridder/xesmf.py:158: in horizontal
    regridder = xe.Regridder(
/usr/share/miniconda3/envs/xcdat_ci/lib/python3.11/site-packages/xesmf/frontend.py:928: in __init__
    super().__init__(
/usr/share/miniconda3/envs/xcdat_ci/lib/python3.11/site-packages/xesmf/frontend.py:385: in __init__
    self.weights = add_nans_to_weights(self.weights)
/usr/share/miniconda3/envs/xcdat_ci/lib/python3.11/site-packages/xesmf/smm.py:229: in add_nans_to_weights
    m.data[krow] = [np.NaN] if m.data[krow] == [] else m.data[krow]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

attr = 'NaN'

    def __getattr__(attr):
        # Warn for expired attributes
        import warnings
    
        if attr == "linalg":
            import numpy.linalg as linalg
            return linalg
        elif attr == "fft":
            import numpy.fft as fft
            return fft
        elif attr == "dtypes":
            import numpy.dtypes as dtypes
            return dtypes
        elif attr == "random":
            import numpy.random as random
            return random
        elif attr == "polynomial":
            import numpy.polynomial as polynomial
            return polynomial
        elif attr == "ma":
            import numpy.ma as ma
            return ma
        elif attr == "ctypeslib":
            import numpy.ctypeslib as ctypeslib
            return ctypeslib
        elif attr == "exceptions":
            import numpy.exceptions as exceptions
            return exceptions
        elif attr == "testing":
            import numpy.testing as testing
            return testing
        elif attr == "matlib":
            import numpy.matlib as matlib
            return matlib
        elif attr == "f2py":
            import numpy.f2py as f2py
            return f2py
        elif attr == "typing":
            import numpy.typing as typing
            return typing
        elif attr == "rec":
            import numpy.rec as rec
            return rec
        elif attr == "char":
            import numpy.char as char
            return char
        elif attr == "array_api":
            raise AttributeError("`numpy.array_api` is not available from "
                                 "numpy 2.0 onwards")
        elif attr == "core":
            import numpy.core as core
            return core
        elif attr == "strings":
            import numpy.strings as strings
            return strings
        elif attr == "distutils":
            if 'distutils' in __numpy_submodules__:
                import numpy.distutils as distutils
                return distutils
            else:
                raise AttributeError("`numpy.distutils` is not available from "
                                     "Python 3.12 onwards")
    
        if attr in __future_scalars__:
            # And future warnings for those that will change, but also give
            # the AttributeError
            warnings.warn(
                f"In the future `np.{attr}` will be defined as the "
                "corresponding NumPy scalar.", FutureWarning, stacklevel=2)
    
        if attr in __former_attrs__:
            raise AttributeError(__former_attrs__[attr])
    
        if attr in __expired_attributes__:
>           raise AttributeError(
                f"`np.{attr}` was removed in the NumPy 2.0 release. "
                f"{__expired_attributes__[attr]}"
            )
E           AttributeError: `np.NaN` was removed in the NumPy 2.0 release. Use `np.nan` instead.

```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -1,6 +1,10 @@
 What's new
 ==========
 
+0.8.6 (unreleased)
+------------------
+* Compatibility with Numpy 2.0 (NaN vs nan) (:pull:`373`) By `Pascal Bourgault <https://github.com/aulemahal>`_.
+
 0.8.5 (2024-04-11)
 ------------------
 * Reverted to the chunking behaviour of xESMF 0.7 for cases where the spatial dimensions are not chunked on the source data. (:pull:`348`) By `Pascal Bourgault <https://github.com/aulemahal>`_.

--- a/xesmf/smm.py
+++ b/xesmf/smm.py
@@ -202,11 +202,11 @@ def apply_weights(weights, indata, shape_in, shape_out):
 
 
 def add_nans_to_weights(weights):
-    """Add NaN in empty rows of the regridding weights sparse matrix.
+    """Add nan in empty rows of the regridding weights sparse matrix.
 
     By default, empty rows in the weights sparse matrix are interpreted as zeroes. This can become problematic
-    when the field being interpreted has legitimate null values. This function inserts NaN values in each row to
-    make sure empty weights are propagated as NaNs instead of zeros.
+    when the field being interpreted has legitimate null values. This function inserts nan values in each row to
+    make sure empty weights are propagated as nans instead of zeros.
 
     Parameters
     ----------
@@ -222,11 +222,11 @@ def add_nans_to_weights(weights):
     # Taken from @trondkr and adapted by @raphaeldussin to use `lil`.
     # lil matrix is better than CSR when changing sparsity
     m = weights.data.to_scipy_sparse().tolil()
-    # replace empty rows by one NaN value at element 0 (arbitrary)
-    # so that remapped element become NaN instead of zero
+    # replace empty rows by one nan value at element 0 (arbitrary)
+    # so that remapped element become nan instead of zero
     for krow in range(len(m.rows)):
         m.rows[krow] = [0] if m.rows[krow] == [] else m.rows[krow]
-        m.data[krow] = [np.NaN] if m.data[krow] == [] else m.data[krow]
+        m.data[krow] = [np.nan] if m.data[krow] == [] else m.data[krow]
     # update regridder weights (in COO)
     weights = weights.copy(data=sps.COO.from_scipy_sparse(m))
     return weights

